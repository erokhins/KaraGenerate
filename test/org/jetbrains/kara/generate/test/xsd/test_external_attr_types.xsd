<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>

    <xsd:simpleType name="externalStrEnumType1">
        <xsd:union>
            <xsd:simpleType>
                <xsd:restriction base="xsd:NMTOKEN">
                    <xsd:enumeration value="value1"/>
                    <xsd:enumeration value="value2"/>
                </xsd:restriction>
            </xsd:simpleType>
            <xsd:simpleType>
                <xsd:restriction base="xsd:string"/>
            </xsd:simpleType>
        </xsd:union>
    </xsd:simpleType>

    <xsd:simpleType name="externalStrEnumType2">
        <xsd:union>
            <xsd:simpleType>
                <xsd:restriction base="xsd:string">
                    <xsd:enumeration value="value-1"/>
                    <xsd:enumeration value="value-2"/>
                </xsd:restriction>
            </xsd:simpleType>
            <xsd:simpleType>
                <xsd:restriction base="xsd:string"/>
            </xsd:simpleType>
        </xsd:union>
    </xsd:simpleType>

    <xsd:simpleType name="externalUnionEnumType">
        <xsd:union>
            <xsd:simpleType>
                <xsd:restriction base="xsd:NMTOKEN">
                    <xsd:enumeration value="value1"/>
                    <xsd:enumeration value="value2"/>
                </xsd:restriction>
            </xsd:simpleType>
        </xsd:union>
    </xsd:simpleType>

    <xsd:simpleType name="externalRestrictionEnumType">
        <xsd:restriction base="xsd:NMTOKEN">
            <xsd:enumeration value="value1"/>
            <xsd:enumeration value="value2"/>
        </xsd:restriction>
    </xsd:simpleType>


    <xsd:element name="complex">
        <xsd:complexType>
            <xsd:attribute name="elementStrEnumType1" type="externalStrEnumType1"/>
            <xsd:attribute name="elementStrEnumType2" type="externalStrEnumType2"/>
            <xsd:attribute name="elementUnionEnumType" type="externalUnionEnumType"/>
            <xsd:attribute name="elementRestrictionEnumType" type="externalRestrictionEnumType"/>
        </xsd:complexType>
    </xsd:element>

</xsd:schema>