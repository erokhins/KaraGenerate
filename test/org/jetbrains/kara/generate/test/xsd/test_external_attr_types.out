AttributeTypeDecl:
   AttributeTypeDecl = (externalRestrictionEnumType, null, enumType)
   Values = [value1, value2]

   AttributeTypeDecl = (externalStrEnumType1, null, strEnumType)
   Values = [value1, value2]

   AttributeTypeDecl = (externalStrEnumType2, null, strEnumType)
   Values = [value-1, value-2]

   AttributeTypeDecl = (externalUnionEnumType, null, enumType)
   Values = [value1, value2]

AttributeDecl:
   AttributeDecl = (elementRestrictionEnumType, null, null})
   AttributeTypeDecl = (externalRestrictionEnumType, null)

   AttributeDecl = (elementStrEnumType1, null, null})
   AttributeTypeDecl = (externalStrEnumType1, null)

   AttributeDecl = (elementStrEnumType2, null, null})
   AttributeTypeDecl = (externalStrEnumType2, null)

   AttributeDecl = (elementUnionEnumType, null, null})
   AttributeTypeDecl = (externalUnionEnumType, null)

AttributeGroupDecl:
ElementDecl:
   ElementDeclaration = (complex, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =
      AttributeDecl = (elementRestrictionEnumType, null, null})
      AttributeTypeDecl = (externalRestrictionEnumType, null)

      AttributeDecl = (elementStrEnumType1, null, null})
      AttributeTypeDecl = (externalStrEnumType1, null)

      AttributeDecl = (elementStrEnumType2, null, null})
      AttributeTypeDecl = (externalStrEnumType2, null)

      AttributeDecl = (elementUnionEnumType, null, null})
      AttributeTypeDecl = (externalUnionEnumType, null)


ElementGroup:
